samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,219 @@
+// Deep learning based Object Detection and Instance Segmentation using Mask R-CNN
+// VIDEO 360 DEMO: https://www.youtube.com/watch?v=0tU8991QgE8
+// Downloads:
+// http://download.tensorflow.org/models/object_detection/mask_rcnn_inception_v2_coco_2018_01_28.tar.gz
+// https://raw.githubusercontent.com/opencv/opencv_extra/master/testdata/dnn/mask_rcnn_inception_v2_coco_2018_01_28.pbtxt
+// More Information : Alessandro de Oliveira Faria (A.K.A. CABELO)- cabelo@opensuse.org
+
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <string.h>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+#include "common.hpp"
+
+std::string keys =
+    "{ help  h     | | Print help message. \nUsage \n\t\t./mask_rcnn --image=logo.jpg \n\t\t ./mask_rcnn --media=teste.mp4}"
+    "{ image m     |<none>| Path to input image file.  }"
+    "{ video v     |<none>| Path to input video file.  }"
+    "{ weights w | ./mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb | The pre-trained weights.  }"
+    "{ textgraph g |./mask_rcnn_inception_v2_coco_2018_01_28.pbtxt | iThe text graph file that has been tuned by the OpenCV’s DNN support group  }"
+    "{ cthr         | .5 | Confidence threshold. }"
+    "{ mthr        | .4 | Mask threshold. }";
+
+
+using namespace cv;
+using namespace dnn;
+using namespace std;
+
+float confThreshold;
+float maskThreshold;
+
+vector<string> classes;
+vector<Scalar> colors;
+
+void drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask);
+
+void postprocess(Mat& frame, const vector<Mat>& outs);
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("Use this script to run object detection using Segmentation using Mask R-CNN in OpenCV.");
```

Comment:
> parser.about("Use this script to run object detection using Segmentation using Mask R-CNN in OpenCV.");

It sounds a bit weird. This sample demonstrates **instance segmentation** network called Mask-RCNN.

Reply:
Done!